File: src/main/java/etl/Etl.java
package main.java.etl;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class Etl {

    public Map<String, Integer> transform(Map<Integer, List<String>> old) {
        Map<String, Integer> result = new HashMap<String, Integer>();
        for (Map.Entry<Integer, List<String>> map : old.entrySet()) {
            int pointKey = map.getKey();
            List<String> list = map.getValue();
            for (String alphas : list) {
                result.put(alphas.toLowerCase(), pointKey);
            }
        }

        return result;
    }
}

File: kotlinapp/src/main/kotlin/com/kotlinapp/algorithms/etl/ETL.kt
package com.kotlinapp.algorithms.etl

/**
 * @author lusinabrian on 31/05/17.
 */
class ETL{
    companion object{
        /**
         * Maps the old data to a new system
         * */
        fun transform(oldMap : Map<Int, List<Char>>) : Map<Char, Int>{
            var result : HashMap<Char, Int> = HashMap()
            for((k, v) in oldMap){
                for (char in v){
                    result.put(char.toLowerCase(), k)
                }
            }
            return result
        }
    }
}
